test/broadcast.jl

```diff
@@ -256,3 +255,10 @@ end
 let foo = [[1,2,3],[4,5,6],[7,8,9]]
     @test max.(foo...) == broadcast(max, foo...) == [7,8,9]
 end
+
+# Issue 17314
```

Comment:
Would be good to test a case where type-inference fails, e.g. `broadcast(x -> x < 0 ? false : x, 1:3)` should return `Int[1,2,3]`.